src/api/app/jobs/stage_requests_job.rb

```diff
@@ -0,0 +1,15 @@
+class StageRequestsJob < ApplicationJob
+  def perform(staging_project, requests)
+    unassigned = []
+    requests.each do |request|
+      # TODO: implement delete requests
+      BranchPackage.new(request: request.number, target_project: staging_project.name, extend_package_names: false).branch
+    rescue
+      unassigned << request
+    end
+
+    # If something goes wrong with branching the package we unassign it again
+    staging_project.staged_requests -= unassigned
```

Comment:
you need to make this as atomic as possible. If you access .staged_requests after a long time doing branches, you will have very strange effects. Not in tests as nothing works concurrent, but in production.